react/features/base/tracks/middleware.js

```diff
@@ -84,14 +85,14 @@ MiddlewareRegistry.register(store => next => action => {
 
         // The TRACK_ADDED action is dispatched when a presenter starts a screenshare. Do not create a local fake
         // screenshare participant when multiple stream is not enabled.
-        const skipCreateFakeScreenShareParticipant = local && !getMultipleVideoSupportFeatureFlag(state);
+        const skipCreateFakeScreenShareParticipant = local && !getMultipleVideoSendingSupportFeatureFlag(state);
 
-        if (getSourceNameSignalingFeatureFlag(state)
+        if (getMultipleVideoSupportFeatureFlag(state)
             && jitsiTrack.getVideoType() === VIDEO_TYPE.DESKTOP
             && !jitsiTrack.isMuted()
             && !skipCreateFakeScreenShareParticipant
         ) {
-            createFakeScreenShareParticipant(store, action);
+            store.dispatch(createFakeScreenShareParticipant(jitsiTrack));
```

Comment:
We are still creating fake SS participants in the middleware but we can have more control over this by moving this to where the `trackAdded` is being dispatched

Reply:
Yes lets move this out of here if we decide not to dispatch actions for fake SS tile creation in tracks middleware when mute changed and videoType changed events are fired.